8373913: Refactor serialization tests to use JUnit

RogerRiggs opened the pull request:
Refactor serialization tests to use JUnit.
Automated conversion for most annotations.
Conditional tests are refactored to use JUnit Enable/DisableIf.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8373913](https://bugs.openjdk.org/browse/JDK-8373913): Refactor serialization tests to use JUnit (**Sub-task** - P4)


### Reviewers
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/28877/head:pull/28877` \
`$ git checkout pull/28877`

Update a local copy of the PR: \
`$ git checkout pull/28877` \
`$ git pull https://git.openjdk.org/jdk.git pull/28877/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 28877`

View PR using the GUI difftool: \
`$ git pr show -t 28877`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/28877.diff">https://git.openjdk.org/jdk/pull/28877.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/28877#issuecomment-3666942742)
</details>

RogerRiggs committed:
JDK-8373913: Refactor serialization tests tests to use JUnit
Automated conversion for most annotations.
Conditional tests are refactored to use JUnit Enable/DisableIf.

liach commented on test/jdk/java/io/Serializable/class/NonSerializableTest.java:
```diff
@@ -96,15 +99,17 @@ public Object[][] provider() {
-    public void test(String[] args) throws Exception {
+    @ParameterizedTest
+    @MethodSource("provider")
+    public void test(String[][] argArray) throws Exception {
```
I recommend changing `provider` to return `List<String[]>` or `Object[]`, instead of hacking the test.

liach commented on test/jdk/java/io/Serializable/serialFilter/SerialFilterTest.java:
```diff
@@ -379,18 +378,15 @@ void testReadResolveToArray(Object array, int length) throws IOException {
-        Assert.assertEquals(
-                filter.checkInput(new FilterValues(arrayClass, value-1, value-1, value-1, value-1)),
-                ObjectInputFilter.Status.UNDECIDED,
+        Assertions.assertEquals(                ObjectInputFilter.Status.UNDECIDED, filter.checkInput(new FilterValues(arrayClass, value-1, value-1, value-1, value-1)),
```
Extra indentation from the auto conversion script

RogerRiggs committed:
Cleanup arglist and fix code style

RogerRiggs committed:
Update to use Stream<List<String>> for parameterized method testing.
Add static to simplify test setup.
Small source changes to remove "C" style array declarations.

AlanBateman commented on test/jdk/java/io/Serializable/serialFilter/CheckArrayTest.java:
```diff
@@ -51,9 +52,10 @@
  * to check array sizes during deserialization via the ObjectInputFilter attached the stream.
  * The filterCheck must be resilent to an InputStream not being available (only the subclass knows).
  */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Looks like 9 other tests are using PER_CLASS, is that because of setup methods and method sources that are instance rather than static methods?

RogerRiggs committed:
Simplification of AssertThrows cases and changing success output to System.err to interleve with JUnit output.

RogerRiggs committed:
Refactored test parameter provider methods to be static.
Removed the @TestInstance annotations.
Added @Serial where it was missing.

RogerRiggs committed:
Merge branch 'master' into 8373913-serialfilter-to-junit

justin-curtis-lu commented on test/jdk/java/io/Serializable/records/RecordClassTest.java:
If we want to remove the class instance lifecycle, can we make `notSerRecordClasses()` static as well?

RogerRiggs committed:
Review suggestions to remove extra spaces, duplicate annotation, and
the last remaining use of PER_CLASS.

justin-curtis-lu reviewed (approved):
LGTM

RogerRiggs commented:
/integrate
